lib/codecs/src/decoding/format/gelf.rs

```diff
@@ -0,0 +1,396 @@
+
+    /// (optional) A long message that can i.e. contain a backtrace
+    pub const FULL_MESSAGE: &str = "full_message";
+
+    /// (optional) Seconds since UNIX epoch with optional decimal places for milliseconds.
+    ///  SHOULD be set by client library. Will be set to the current timestamp (now) by the server if absent.
+    pub const TIMESTAMP: &str = "timestamp";
+
+    /// (optional) The level equal to the standard syslog levels. default is 1 (ALERT).
+    pub const LEVEL: &str = "level";
+
+    /// (optional) (deprecated) Send as additional field instead.
+    pub const FACILITY: &str = "facility";
+
+    /// (optional) (deprecated) The line in a file that caused the error (decimal). Send as additional field instead.
+    pub const LINE: &str = "line";
+
+    /// (optional) (deprecated) The file (with path if you want) that caused the error. Send as additional field instead.
+    pub const FILE: &str = "file";
+}
+pub use gelf_fields::*;
+
+/// Config used to build a `GelfDeserializer`.
+#[derive(Debug, Clone, Default, Deserialize, Serialize)]
+pub struct GelfDeserializerConfig;
+
+impl GelfDeserializerConfig {
+    /// Build the `GelfDeserializer` from this configuration.
+    pub fn build(&self) -> GelfDeserializer {
+        GelfDeserializer::default()
+    }
+
+    /// Return the type of event built by this deserializer.
+    pub fn output_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    /// The schema produced by the deserializer.
+    pub fn schema_definition(&self) -> schema::Definition {
+        schema::Definition::empty()
+            .with_field(VERSION, Kind::bytes(), None)
+            .with_field(HOST, Kind::bytes(), None)
+            .with_field(SHORT_MESSAGE, Kind::bytes(), None)
+            .optional_field(FULL_MESSAGE, Kind::bytes(), None)
+            .optional_field(TIMESTAMP, Kind::timestamp(), None)
+            .optional_field(LEVEL, Kind::integer(), None)
+            .optional_field(FACILITY, Kind::bytes(), None)
+            .optional_field(LINE, Kind::integer(), None)
+            .optional_field(FILE, Kind::bytes(), None)
+            // Every field with an underscore (_) prefix will be treated as an additional field.
+            // Allowed characters in field names are any word character (letter, number, underscore), dashes and dots.
+            // Libraries SHOULD not allow to send id as additional field ( _id). Graylog server nodes omit this field automatically.
+            .unknown_fields(Kind::bytes())
+    }
+}
+
+/// Deserializer that builds an `Event` from a byte frame containing a GELF log
+/// message.
+#[derive(Debug, Clone)]
+pub struct GelfDeserializer {
+    regex: Regex,
+}
+
+impl Default for GelfDeserializer {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl GelfDeserializer {
+    /// Create a new GelfDeserializer
+    pub fn new() -> GelfDeserializer {
+        GelfDeserializer {
+            regex: Regex::new(r"^_[\w\.\-]*$").unwrap(),
```

Comment:
Dump this into a `once_cell` like we did in the encoder? I don't know if it makes a noticeable difference...

Reply:
Good call. I believe it will because it'll be static that way so we would improve performance on creation(s) of the `GelfDeserializer`.

Reply:
I'm thinking we'd only do that once (per component using it) at startup so I doubt it's a hotspot, but it also doesn't hurt (nor does it hurt to keep the same pattern).